### PR TITLE
Gsm8k lora

### PR DESCRIPTION
**Pull request by dlwh**

### CODE REVIEW

File: examples/gsm8k-lora/gsm8k_lora.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+            else:
+                loss_mask = 1 - hax.nn.one_hot(-1, Pos, dtype=jax.numpy.float32)
+
+            yield LmExample(input_ids, loss_mask)
+
+
+def mk_dataset(config: TrainArgs, tokenizer: transformers.PreTrainedTokenizerBase):
+    dataset = WrappedHFDataset("gsm8k", split="train", name="main")
+
+    def preprocess(batch):
```
Comment (Ivan-Zhou) [inline examples/gsm8k-lora/gsm8k_lora.py:97]: Not sure if I miss anything else. Is this data formatting the main place where it differs from the existing alpaca and alpaca_lora?

Comment (dlwh) [reply examples/gsm8k-lora/gsm8k_lora.py:97]: yeah everything else is cosmetic

Comment (dlwh) [reply examples/gsm8k-lora/gsm8k_lora.py:97]: i added the ability to choose whether to mask inputs for the loss calculation, i guess
